src/dataprotection/azext_dataprotection/manual/helpers.py

```diff
@@ -92,3 +92,79 @@ def get_backup_frequency_from_time_interval(repeating_time_intervals):
 def get_tagging_priority(name):
     priorityMap = {"Default": 99, "Daily": 25, "Weekly": 20, "Monthly": 15, "Yearly": 10}
     return priorityMap[name]
+
+
+def truncate_id_using_scope(arm_id, scope):
+    scope_map = {"Subscription": 3, "ResourceGroup": 5, "Resource": None}
+    return "/".join(arm_id.split("/")[:scope_map[scope]])
+
+
+def get_sub_id_from_arm_id(arm_id):
+    return truncate_id_using_scope(arm_id, "Subscription")
+
+
+def get_rg_id_from_arm_id(arm_id):
+    return truncate_id_using_scope(arm_id, "ResourceGroup")
```

Comment:
You can leverage `is_valid_resource_id` and `parse_resource_id` from `azure.mgmt.core.tools`